### ISSUE TITLE
change is not changing bound variable names

### ISSUE THREAD

**Issue by PatrickMassot** [leanprover-community/lean#95]
The following call to change is doing nothing.
```lean
example (P : ℕ → Prop) (h : ∀ n, P n) : true :=
begin
  change ∀ k, P k at h,
  trivial
end
```

**Comment by cipher1024** [comment]
That's on purpose. When the expressions are alpha equivalent, the tactic doesn't do anything. We could use equality instead of alpha equivalence I suppose.

**Comment by cipher1024** [comment]
The easiest thing to do if you need it for a tactic is to create `change'` and when we decide to move primed tactic to Lean 3.5

**Comment by cipher1024** [comment]
This should do it:

```lean
meta def change' (e : parse texpr) : tactic unit :=
do e <- to_expr e,
   v <- mk_meta_var e,
   exact v,
   gs <- get_goals,
   set_goals $ v :: gs
```

**Comment by cipher1024** [comment]
Btw, in this branch, `change` should have the behavior you expect: https://github.com/leanprover-community/lean/tree/change-core-alpha

**Comment by PatrickMassot** [comment]
@robertylewis, what do you think? Should we put this `change'` in mathlib or wait for Lean 3.5.1c? @cipher1024 would you be happy with this fix in Lean?

**Comment by robertylewis** [comment]
> That's on purpose. When the expressions are alpha equivalent, the tactic doesn't do anything. We could use equality instead of alpha equivalence I suppose.

This suggestion sounds right to me. I'd support merging the `change-core-alpha` commit. If mathlib is moving to 3.5c fairly soon, I don't think it's necessary to add and then remove `change'`.

**Comment by cipher1024** [comment]
Ok, I'll add a test and get it reviewed

**Comment by Vtec234** [comment]
Can this be closed now that the new `change` has been merged?

**Comment by robertylewis** [comment]
> Can this be closed now that the new `change` has been merged?

Yes, seems fair. This was fixed in f89e46e6b2f6656631384d080d430d4776f11055 (thanks @cipher1024 !).
